### PR TITLE
GH-37232: [MATLAB] Add `arrow.type.Time64Type` class and `arrow.time64` construction function 

### LINKED ISSUE #37232: [MATLAB] Add `arrow.type.Time64Type` class and `arrow.time64` construction function
### Describe the enhancement requested

To add support for `arrow.array.Time64Array`, we will need to add support for a `arrow.type.Time64Type` class and associated `arrow.time64` construction function to the MATLAB interface.

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

To add support for `arrow.array.Time64Array`, we will need to add support for a `arrow.type.Time64Type` class and associated `arrow.time64` construction function to the MATLAB interface.

### What changes are included in this PR?

1. New `arrow.type.Time64Type` class.
2. New `arrow.time64` construction function that returns an `arrow.type.Time64Type` instance.

**Example**

```matlab
>> type = arrow.time64(TimeUnit="Nanosecond")

type = 

  Time64Type with properties:

          ID: Time64
    TimeUnit: Nanosecond

>> class(type)

ans =

    'arrow.type.Time64Type'
```

### Are these changes tested?

Yes.

1. Added a new test class `tTime64Type`.
2. Added new tests for `Time64` type ID enumeration to `tID`.

### Are there any user-facing changes?

Yes.

1. `arrow.time64` construction function
3. `arrow.type.Time64Type` class

### Future Directions

1. #37229
2. #37230 
3. Add `arrow.array.Time64Array` class  

* Closes: #37232

### CODE REVIEW

File: matlab/test/arrow/type/tID.m [hunk 1]
```diff
@@ -43,7 +43,8 @@ function CastToUInt64(testCase)
                 ID.Float64,   12, ...
                 ID.String,    13, ...
                 ID.Timestamp, 18, ...
-                ID.Time32,    19  ...
+                ID.Time32,    19,  ...
```
Comment (kevingurney) [inline matlab/test/arrow/type/tID.m:46]: I think this ellipsis is out of line.

**Review by kevingurney — CHANGES_REQUESTED** [review]
Thanks for the pull request!

File: matlab/test/arrow/type/tID.m (hunk 1 above)
Comment (sgilmore10) [reply matlab/test/arrow/type/tID.m]: good catch!

**Review by kevingurney — APPROVED** [review]

**Comment by kevingurney** [comment]
+1

**Pull request merged**